pyslim/slim_metadata.py

```diff
@@ -11,6 +13,431 @@
+                    "type": "number",
+                    "description": "The frequency with which males in this subpopulation reproduce clonally (for WF models).",
+                    "binaryFormat": "d",
+                    "index": 4},
+                "sex_ratio": {
+                    "type": "number",
+                    "description": "This subpopulation's sex ratio (for WF models).",
+                    "binaryFormat": "d",
+                    "index": 5},
+                "bounds_x0": {
+                    "type": "number",
+                    "description": "The minimum x-coordinate in this subpopulation.",
+                    "binaryFormat": "d",
+                    "index": 6},
+                "bounds_x1": {
+                    "type": "number",
+                    "description": "The maximum x-coordinate in this subpopulation.",
+                    "binaryFormat": "d",
+                    "index": 7},
+                "bounds_y0": {
+                    "type": "number",
+                    "description": "The minimum y-coordinate in this subpopulation.",
+                    "binaryFormat": "d",
+                    "index": 8},
+                "bounds_y1": {
+                    "type": "number",
+                    "description": "The maximum y-coordinate in this subpopulation.",
+                    "binaryFormat": "d",
+                    "index": 9},
+                "bounds_z0": {
+                    "type": "number",
+                    "description": "The minimum z-coordinate in this subpopulation.",
+                    "binaryFormat": "d",
+                    "index": 10},
+                "bounds_z1": {
+                    "type": "number",
+                    "description": "The maximum z-coordinate in this subpopulation.",
+                    "binaryFormat": "d",
+                    "index": 11},
+                "migration_records": {
+                    "type": "array",
+                    "index": 13,
+                    "items": {
+                        "type": "object",
+                        "properties": {
+                            "source_subpop": {
+                                "type": "integer",
+                                "description": "The ID of the subpopulation migrants come from (in WF models).",
+                                "binaryFormat": "i",
+                                "index": 1},
+                            "migration_rate": {
+                                "type": "number",
+                                "description": "The fraction of children in this subpopulation that are composed of 'migrants' from the source subpopulation (in WF models).",
+                                "binaryFormat": "d",
+                                "index": 2}
+                        },
+                        "required": ["source_subpop", "migration_rate"],
+                        "additionalProperties": False
+                    }
+                }
```

Comment:
How is the length of this array encoded? Is it relying on the default array length type being `Q`? Might be nice to be explicit about that?

Reply:
[Here's](https://github.com/tskit-dev/pyslim/blob/master/pyslim/slim_metadata.py#L399) a handy reference. And, [here's](https://github.com/tskit-dev/pyslim/blob/master/pyslim/slim_metadata.py#L465) how I was doing the decoding. (I can get rid of that code now, though!) Oh, I see - I just need to put an `"arrayLengthFormat" : "Q"` entry in here, huh?

Reply:
oh, weird - it looks like I should have done `"arrayLengthFormat" : "i"` - how did this even work?

Reply:
uh-oh, but the documentation says that `i` is not an option, only `I`!

Reply:
I'll do `"arrayLengthFormat" : "I"` - https://github.com/tskit-dev/tskit/issues/769

Reply:
... and, we've changed the specification in C++ to `uint32_t`, since @bhaller says that's OK. =)

Reply:
Phew, glad that worked out - sorry that the docs had the default wrong - will fix.